ktor-io/common/src/io/ktor/utils/io/core/AbstractInput.kt

```diff
@@ -44,14 +68,13 @@ public abstract class AbstractInput(
     /**
      * Current head chunk reference
      */
-    private final var __head: ChunkBuffer by shared(head)
     private final var _head: ChunkBuffer
-        get() = __head
+        get() = state.head
```

Comment:
I wonder why didn't we seriously slow down because of this? We added an indirection here (`by shared`) so this should be much slower, isn't it?

Reply:
We removed allocation at least, and indirection is in the final field